tests/test_separation.py

```diff
@@ -68,6 +68,52 @@ def __unit_test_separation_function(metric):
     nose.tools.assert_raises(ValueError, metric, sources, sources)
 
 
+def __unit_test_sources_framewise_function(metric):
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        # First, test for a warning on empty audio data
+        metric(np.array([]), np.array([]), 400, 200)
+        assert len(w) == 2
+        assert issubclass(w[-1].category, UserWarning)
+        assert str(w[-1].message) == ("estimated_sources is empty, "
+                                      "should be of size (nsrc, nsample).  "
+                                      "sdr, sir, sar, and perm will all be "
+                                      "empty np.ndarrays")
+        # And that the metric returns empty arrays
+        assert np.allclose(
+            metric(np.array([]), np.array([]), 40, 20), np.array([])
+        )
+
+    # Test for error when there is a silent reference/estimated source
```

Comment:
Yeah, you're right... The code duplication is bad enough for having 2 unit tests, but it will get much worse if/when we add image and image framewise evaluation.. Having multiple test cases allows them to be reused across the eventual 4 functions. Good catch!